controller/src/main/java/org/jboss/as/controller/operations/global/ReadFeatureDescriptionHandler.java

```diff
@@ -497,6 +500,31 @@ private void processObjectAttribute(final ModelNode parentFeature, final Immutab
                 param.get(TYPE).set("List<String>");
             }
             params.add(param);
+            final ModelNode attrDescription = attr.getNoTextDescription(false);
+            if (attrDescription.hasDefined(CAPABILITY_REFERENCE)) {
+                ModelNode capability = new ModelNode();
+                String capabilityName = attrDescription.get(CAPABILITY_REFERENCE).asString();
+                final String baseName;
+                if (capabilityName.indexOf('$') > 0) {
+                    baseName = capabilityName.substring(0, capabilityName.indexOf('$') - 1);
+                } else {
+                    baseName = capabilityName;
+                }
+                CapabilityRegistration<?> capReg = getCapability(new CapabilityId(baseName, CapabilityScope.GLOBAL));
+                if (capReg == null || capReg.getCapability().isDynamicallyNamed() && capabilityName.indexOf('$') <= 0) {
+                    capabilityName = baseName + ".$" + paramName;
+                }
+                capability.get(OPTIONAL).set(attrDescription.hasDefined(NILLABLE) && attrDescription.get(NILLABLE).asBoolean());
+                if (isProfile) {
+                    capabilityName = PROFILE_PREFIX + capabilityName;
+                }
+                capability.get(NAME).set(capabilityName);
+                required.add(capability);
+            }
+        }
```

Comment:
Why doesn't this block do everything that is done in the block starting at https://github.com/bstansberry/wildfly-core/blob/master/controller/src/main/java/org/jboss/as/controller/operations/global/ReadFeatureDescriptionHandler.java#L736?  Except the 'filteredOut' bit at https://github.com/bstansberry/wildfly-core/blob/master/controller/src/main/java/org/jboss/as/controller/operations/global/ReadFeatureDescriptionHandler.java#L750. That 'filteredOut' bit BTW smells funny. AFAICT if filteredOut==true the entire for loop starting at L735 can be skipped as no non-local state is changed in the for loop.